contracts/main/stablecoin-core/AdminControls.sol

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+import "../interfaces/IAccessControlConfig.sol";
+import "../interfaces/ICollateralPoolConfig.sol";
+import "../interfaces/IPausable.sol";
+import "../interfaces/IBookKeeper.sol";
+
+contract AdminControls is OwnableUpgradeable {
+    address public bookKeeper;
+    address public liquidationEngine;
+    address public priceOracle;
+    address public positionManager;
+    address public stableSwapModule;
+    address public systemDebtEngine;
+    address public flashMintModule;
+    address public stablecoinAdapter;
+
+    event LogPauseProtocol();
+    event LogUnpauseProtocol();
+
+    modifier onlyOwnerOrGov() {
+        IAccessControlConfig _accessControlConfig = IAccessControlConfig(IBookKeeper(bookKeeper).accessControlConfig());
+        require(
+            _accessControlConfig.hasRole(_accessControlConfig.OWNER_ROLE(), msg.sender) ||
+                _accessControlConfig.hasRole(_accessControlConfig.GOV_ROLE(), msg.sender),
+            "!(ownerRole or govRole)"
+        );
+        _;
+    }
+
+    function initialize(
+        address _bookKeeper,
+        address _liquidationEngine,
+        address _priceOracle,
+        address _positionManager,
+        address _stableSwapModule,
+        address _systemDebtEngine,
+        address _flashMintModule,
+        address _stablecoinAdapter
+    ) external initializer {
+        bookKeeper = _bookKeeper;
+        liquidationEngine = _liquidationEngine;
+        priceOracle = _priceOracle;
+        positionManager = _positionManager;
+        stableSwapModule = _stableSwapModule;
+        flashMintModule = _flashMintModule;
+        systemDebtEngine = _systemDebtEngine;
+        stablecoinAdapter = _stablecoinAdapter;
+    }
+
+    function pauseProtocol() external onlyOwnerOrGov {
+        IPausable(bookKeeper).pause();
+        IPausable(positionManager).pause();
+        IPausable(liquidationEngine).pause();
+        IPausable(systemDebtEngine).pause();
+        IPausable(stableSwapModule).pause();
```

Comment:
I think StableSwapModule should keep going although the protocol is paused to ensure FXD pegging at all time.

Reply:
agree, will change

Reply:
Thank you bossue (_ _ )

Reply:
fixed